src/HappyCode.NetCoreBoilerplate.Api/Program.cs

```diff
@@ -1,19 +1,51 @@
+using System;
 using Autofac.Extensions.DependencyInjection;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+using Serilog.Core;
 
 namespace HappyCode.NetCoreBoilerplate.Api
 {
     public static class Program
     {
         public static void Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            Log.Logger = CreateLogger();
+
+            try
+            {
+                Log.Logger.Information("Starting up");
+                using var webHost = CreateWebHostBuilder(args).Build();
+                webHost.Run();
+            }
+            catch (Exception ex)
+            {
+                Log.Logger.Fatal(ex, "Application start-up failed");
+                Log.CloseAndFlush();
+                throw;
+            }
         }
 
         public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
+                .UseSerilog()
                 .ConfigureServices(services => services.AddAutofac())
                 .UseStartup<Startup>();
+
+        private static Logger CreateLogger()
```

Comment:
Please move this method to src\HappyCode.NetCoreBoilerplate.Api\Infrastructure\Configurations\SerilogConfigurator.cs

Reply:
Done